src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -526,6 +526,44 @@ def generate_task(self) -> 'Task':
+                        elif param.annotation == bool and isinstance(arg_value, str):
+                            casted_args[param_name] = arg_value.lower() in ('true', '1', 'yes', 'on')
+                        else:
+                            casted_args[param_name] = arg_value
+                    else:
+                        casted_args[param_name] = arg_value
+                else:
+                    casted_args[param_name] = arg_value
+            
+            return casted_args
+        except Exception as e:
+            logging.debug(f"Type casting failed for {function_name}: {e}")
```

Ruff (0.11.9)

Undefined name `function_name`

(F821)